rotkehlchen/chain/manager.py

```diff
@@ -771,23 +793,33 @@ def query_polkadot_balances(
             total_balance += balance
         self.totals.assets[A_DOT] = total_balance
 
-    def sync_btc_accounts_with_db(self) -> None:
-        """Call this function after having deleted BTC accounts from the DB to
+    def sync_btc_accounts_with_db(
+        self,
+        blockchain: Literal[SupportedBlockchain.BITCOIN, SupportedBlockchain.BITCOIN_CASH],
+    ) -> None:
+        """Call this function after having deleted BTC/BCH accounts from the DB to
         sync the chain manager's balances and accounts with the DB
 
         For example this is called after removing an xpub which deletes all derived
         addresses from the DB.
         """
-        db_btc_accounts = self.database.get_blockchain_accounts().btc
+        db_btc_accounts = getattr(
+            self.database.get_blockchain_accounts(),
+            blockchain.value.lower(),
+        )
         accounts_to_remove = []
-        for btc_account in self.accounts.btc:
-            if btc_account not in db_btc_accounts:
-                accounts_to_remove.append(btc_account)
+        for account in getattr(self.accounts, blockchain.value.lower()):
+            if account not in db_btc_accounts:
+                accounts_to_remove.append(account)
 
-        balances_mapping = get_bitcoin_addresses_balances(accounts_to_remove)
+        get_blockchain_balances = {
+            'BCH': get_bitcoin_cash_addresses_balances,
+            'BTC': get_bitcoin_addresses_balances,
+        }
+        balances_mapping = get_blockchain_balances[blockchain.value](accounts_to_remove)
```

Comment:
Hmmm what are you doing here?

Why are you calling both `get_bitcoin_cash_addresses_balances` and `get_bitcoin_addresses_balances`?

In the end in line 819 you only use one of those. Why go to the extra trouble and not just the appropriate one depending on the passed `blockchain` value?